src/org/tensorflow/yolo/view/ClassifierActivity.java

```diff
@@ -136,11 +184,65 @@ private void renderAdditionalInformation(final Canvas canvas) {
             }
         }
 
-        lines.add("Frame: " + previewWidth + "x" + previewHeight);
-        lines.add("View: " + canvas.getWidth() + "x" + canvas.getHeight());
-        lines.add("Rotation: " + sensorOrientation);
-        lines.add("Inference time: " + lastProcessingTimeMs + "ms");
+// shimatani
+        // lines.add("Frame: " + previewWidth + "x" + previewHeight);
+        // lines.add("View: " + canvas.getWidth() + "x" + canvas.getHeight());
+        // lines.add("Rotation: " + sensorOrientation);
+        // lines.add("Inference time: " + lastProcessingTimeMs + "ms");
+
+        Log.d(LOGGING_TAG, String.format("検出名: %s", nowRecognizedClass));
+        lines.add("検出名: " + nowRecognizedClass);
+
+        String msgInf = makeTts(nowRecognizedClass);
+
+        lines.add(msgInf1);
+        lines.add(msgInf2);
 
         borderedText.drawLines(canvas, 10, 10, lines);
     }
+
+    private void run() {
+        final long startTime = SystemClock.uptimeMillis();
+        final List<Recognition> results = recognizer.recognizeImage(croppedBitmap);
+        lastProcessingTimeMs = SystemClock.uptimeMillis() - startTime;
+        overlayView.setResults(results);
+
+        // shimatani
+        Log.d(LOGGING_TAG, "Debug: runInBackground()");
+
+        if (!(results.isEmpty())) {
+            nowResult = results.get(0).getTitle();
+            Log.d(LOGGING_TAG, String.format("Find: %s", nowResult));
+
+            if (!(results.isEmpty()) && lastResult.equals(nowResult)) {
+                matchCount++;
+                if (matchCount > 3) {
+                    // match 4 times
+                    matchCount = 0;
+
+                    if (!(results.isEmpty() || lastRecognizedClass.equals(nowResult))) {
+                        nowRecognizedClass = nowResult;
+                        Log.d(LOGGING_TAG, "Match 4 times: " + nowRecognizedClass);
+                        lastRecognizedClass = nowRecognizedClass;
+
+                        tts = makeTts(nowRecognizedClass);
+                        Log.d(LOGGING_TAG, "makeTts(): " + tts);
+                        if (!(tts.equals(""))) {
```

Comment:
If the default value for tts is null then you don't need to check its value with `equals`.

Reply:
Depending on the specifications of the `makeTts()`function, the return value may be `null`. In this case, I made it so as not to talk. (I made sure that the `speak2()` function does not get an error)